### ISSUE TITLE
 .fit() hangs when using DDP with relative imports in main code

### ISSUE THREAD

**Issue by Shreeyak** [PyTorchLightning/pytorch-lightning#4243]
## 🐛 Bug

My training script is a module within a package. If my module uses relative imports and `ddp` backend, it throws an error about relative imports and hangs. Using `ddp_spawn` backend and relative imports works as expected.
The process becomes unresponsive even to Ctrl-C and I have to kill it and its subprocesses by PID.
```shells script
Traceback (most recent call last):
  File "/home/shrek/pytorch-lightning-segmentation-lapa/seg_lapa/train.py", line 13, in <module>
    from .networks.deeplab.deeplab import DeepLab
ImportError: attempted relative import with no known parent package
```

Using absolute imports instead of relative imports fixes the issue.

## Please reproduce using the BoringModel
Not sure how to reproduce on the BoringModel, since this bug involves creating a python package and importing modulest.

### To Reproduce
Create a simple python package and import module using relative imports
```python
# Relative Imports - hangs
from .networks.deeplab.deeplab import DeepLab
from .loss_func import CrossEntropy2D
from .dataloaders import LapaDataset, DatasetSplit

# Absolute Imports - works
# from seg_lapa.networks.deeplab.deeplab import DeepLab
# from seg_lapa.loss_func import CrossEntropy2D
# from seg_lapa.dataloaders import LapaDataset, DatasetSplit

trainer = pl.Trainer(gpus=[0, 1], distributed_backend="ddp", num_nodes=1, precision=32)
trainer.fit(model, train_loader)
```

### Expected behavior

I would expect `ddp` to work regardless of absolute/relative imports. The model should start training per normal.

### Environment

```
* CUDA:
	- GPU:
		- GeForce GTX 1080 Ti
		- GeForce GTX 1080 Ti
	- available:         True
	- version:           10.2
* Packages:
	- numpy:             1.19.2
	- pyTorch_debug:     False
	- pyTorch_version:   1.6.0
	- pytorch-lightning: 1.0.2
	- tqdm:              4.50.2
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- ELF
	- processor:         x86_64
	- python:            3.8.5
	- version:           #109-Ubuntu SMP Fri Jun 19 11:33:10 UTC 2020

```

 - PyTorch Version (e.g., 1.0): 1.6.0
 - OS (e.g., Linux): Ubuntu 18.04
 - How you installed PyTorch (`conda`, `pip`, source): pip install within conda env
 - Build command you used (if compiling from source): N/A
 - Python version: 3.8.5
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by awaelchli** [comment]
@Shreeyak I don't really know how to solve this problem. In the docs it is mentioned that DDP does not work without a root  package
https://pytorch-lightning.readthedocs.io/en/stable/multi_gpu.html#distributed-data-parallel

For running the script with the "-m" option, we have a similar feature request here: #3689 (maybe this is what you need?)

**Comment by Shreeyak** [comment]
Thanks for looking into this @awaelchli. Could you please explain what is mean by "root package"? 

To clarify, I'm able to run DDP with the command `python -m seg_lapa.train`. `.fit()` works fine, so long as the imports are absolute imports. I have yet to test `.test()`.

`seg_lapa` is my package. Here's my dir structure:
```
.
├── LICENSE
├── lightning_logs
├── README.md
├── scripts
├── seg_lapa
│   ├── train.py
│   ├── ...
├── seg_lapa.egg-info
├── setup.cfg
├── setup.py
```

**Comment by awaelchli** [comment]
> Could you please explain what is mean by "root package"

Tbh, I don't know sorry, I don't know what it means. 

I'm not familiar with relative imports, I am trying to reproduce and I even get the same error message locally when training on cpu. I took the [pl_examples/basic_examples/image_classifier.py](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/basic_examples/image_classifier.py) and took out the Backbone, placing it in a new file:

```
├── pl_examples
│   ├── basic_examples
│         ├── backbone.py
│         ├── image_classifier.py
│         ├── ... 
```
and in image_classifier I have 

```python
from .backbone import Backbone
```
this is the only change I made to this example code.
Then:

```bash
cd pl_examples/basic_examples
python image_classifier.py 

ImportError: attempted relative import with no known parent package
```

You report that it is only for ddp, so I must be doing something wrong here.

**Comment by JohnnyRisk** [comment]
I have this same problem using relative imports. I try to use -m flag and the process hangs.

**Comment by opringle** [comment]
Having the same issue. Training on GCP Ai Platform with the following project structure - https://cloud.google.com/ai-platform/training/docs/packaging-trainer#project-structure. Using ddp accelerator causes an import error. Without ddp works fine.

**Comment by PonyCheng** [comment]
Same issue...Did anyone solve the issue?

**Comment by nexplor** [comment]
This should have been solved way back in #4243
